examples/gno.land/p/demo/teritori/dao_interfaces/messages_registry.gno

```diff
@@ -0,0 +1,152 @@
+package dao_interfaces
+
+import (
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/json"
+)
+
+type MessagesRegistry struct {
+	handlers *avl.Tree
+}
+
+func NewMessagesRegistry() *MessagesRegistry {
+	registry := &MessagesRegistry{handlers: avl.NewTree()}
+	registry.Register(NewRegisterHandlerExecutableMessageHandler(registry))
+	registry.Register(NewRemoveHandlerExecutableMessageHandler(registry))
+	return registry
+}
+
+func (r *MessagesRegistry) Register(handler MessageHandler) {
+	r.handlers.Set(handler.Type(), handler)
+}
+
+func (r *MessagesRegistry) Remove(t string) {
+	r.handlers.Remove(t)
+}
+
+func (r *MessagesRegistry) MessagesFromJSON(slice []*json.Node) []ExecutableMessage {
+	msgs := make([]ExecutableMessage, len(slice))
+	for i, elem := range slice {
+		messageType := json.Must(elem.GetKey("type")).MustString()
+		payload := json.Must(elem.GetKey("payload"))
+		h, ok := r.handlers.Get(messageType)
+		if !ok {
+			panic("invalid ExecutableMessage: invalid message type")
+		}
+		instance := h.(MessageHandler).Instantiate()
+		instance.FromJSON(payload)
+		msgs[i] = instance
+	}
+	return msgs
+}
+
+func (r *MessagesRegistry) Execute(msg ExecutableMessage) {
+	h, ok := r.handlers.Get(msg.Type())
+	if !ok {
+		panic("invalid ExecutableMessage: invalid message type")
+	}
+	h.(MessageHandler).Execute(msg)
+}
+
+func (r *MessagesRegistry) ExecuteMessages(msgs []ExecutableMessage) {
+	for _, msg := range msgs {
+		r.Execute(msg)
+	}
+}
+
+func (r *MessagesRegistry) Render() string {
+	s := ""
+	r.handlers.Iterate("", "", func(key string, value interface{}) bool {
+		s += "- " + value.(MessageHandler).Type() + "\n"
+		return false
+	})
+	return s
+}
```

Comment:
For the same reason, it would be good to use `strings.Builder` here as well.
```suggestion
func (r *MessagesRegistry) Render() string {
    var sb strings.Builder

    r.handlers.Iterate("", "", func(key string, value interface{}) bool {
        sb.WriteString("- ")
        sb.WriteString(value.(MessageHandler).Type())
        sb.WriteString("\n")
        return false
    })

    return sb.String()
}
```